subworkflows/local/genome_assembly.nf

```diff
@@ -14,77 +16,105 @@ workflow ASSEMBLY {
 
     if(params.hifiasm_mode == 'hifi-only') {
 
-        // Why is this working?
-        x = Channel.of([]).toList()
-        y = ch_reads.combine(x).combine(x)
-
-        HIFIASM ( y, [], []) // [ [meta], fastq ]
-    } else if(params.hifiasm_mode == 'trio-binning') {
+        ch_reads
+            .groupTuple()
+            .map{ meta, reads ->
+                [ meta, reads, [], [] ]
+            }
+            .set { hifiasm_in }
 
-        // TODO: Multiple trios with different parents may not work! But this is not checked in PED either
+        HIFIASM ( hifiasm_in, [], [] )
+        ch_versions = ch_versions.mix(HIFIASM.out.versions)
 
-        ch_reads
-            .map{it[0]}
-            .combine(ch_reads
-                .map{ meta, reads -> [meta.id, reads]
-                }
-            )
-            .branch{
-                kid: it[0]['id']          == it[1]
-                mom: it[0]['maternal_id'] == it[1]
-                dad: it[0]['paternal_id'] == it[1]
+    } else if(params.hifiasm_mode == 'trio-binning') {
+        // TODO: Multiple trios with different parents may not work?
+        ch_reads.groupTuple()
+            .map{ meta, reads -> meta } // Takes meta, then
+            // combine to create all possible combinations of [ meta, meta ]
+            // but keep only the id of the second meta [ meta, meta.id ]
+            .combine(ch_reads.groupTuple().map{ meta, reads -> [meta.id, reads] })
+            // This creates [ meta, meta.id, reads ], where
```

Comment:
```suggestion
            // This creates [ meta.id, reads ], where
```

Reply:
?

Reply:
No, we have `[ meta ]`, we combine it with `[ meta.id, reads ]` to create `[ meta, meta.id, reads ]`, or maybe rather `[ meta, [meta.id, reads] ]`.

Sorry, this whole logic (together with the dipcall module) is really the eyesore of the pipeline. I wish to fix it for real, but it's not prio atm.

Reply:
I agree, it is not necessary for this release, but use this one then: [ meta, [meta.id, reads] ]